### ISSUE TITLE
JSON serialize fails - could not find implicit evidence

### ISSUE THREAD

**Issue by MLnick** [sksamuel/avro4s#38]
Am I doing something daft? From the example:

```scala
object AvroTest extends App {
  import com.sksamuel.avro4s.AvroOutputStream

  case class Composer(name: String, birthplace: String, compositions: Seq[String])
  val ennio = Composer("ennio morricone", "rome", Seq("legend of 1900", "ecstasy of gold"))

  val baos = new ByteArrayOutputStream()
  val output = AvroOutputStream.json[Composer](baos)
  output.write(ennio)
  output.close()
  println(baos.toString("UTF-8"))
}
```

gives

```
Error:(14, 47) could not find implicit value for evidence parameter of type com.sksamuel.avro4s.ToRecord[AvroTest.Composer]
  val output = AvroOutputStream.json[Composer](baos)
                                              ^

Error:(14, 47) not enough arguments for method json: (implicit evidence$5: com.sksamuel.avro4s.SchemaFor[AvroTest.Composer], implicit evidence$6: com.sksamuel.avro4s.ToRecord[AvroTest.Composer])com.sksamuel.avro4s.AvroJsonOutputStream[AvroTest.Composer].
Unspecified value parameter evidence$6.
  val output = AvroOutputStream.json[Composer](baos)
                                              ^
```

using sbt:

```
 sbt compile
[info] Loading project definition from /Users/nick/workspace/scala/avro4s-example/project
[info] Set current project to avro4s-example (in build file:/Users/nick/workspace/scala/avro4s-example/)
[info] Updating {file:/Users/nick/workspace/scala/avro4s-example/}avro4s-example...
[info] Resolving jline#jline;2.12.1 ...
[info] Done updating.
[info] Compiling 1 Scala source to /Users/nick/workspace/scala/avro4s-example/target/scala-2.11/classes...
[error] /Users/nick/workspace/scala/avro4s-example/src/main/scala-2.11/AvroTest.scala:12: could not find implicit value for evidence parameter of type com.sksamuel.avro4s.ToRecord[AvroTest.Composer]
[error]   val output = AvroOutputStream.json[Composer](baos)
[error]                                               ^
[error] one error found
[error] (compile:compileIncremental) Compilation failed
[error] Total time: 5 s, completed 05 Jul 2016 4:28:07 PM
```

**Comment by sksamuel** [comment]
I get same error when I try a package outside of com.sksamuel.avro, which all the tests are using. Strange.

**Comment by sksamuel** [comment]
It seems to work if I undo the recent caching changes. Perhaps something in there isn't being properly scoped @jeroentervoorde

**Comment by MLnick** [comment]
yeah I tried 1.4.3 and it does work.

**Comment by jeroentervoorde** [comment]
Yes. Problem was in ToRecord macro not using a fully qualified name for SchemaFor
Sorry for not noticing this before. I had a import com.sksamuel.avro4s._ in my test code.

I made a PR (https://github.com/sksamuel/avro4s/pull/39) to fix this.

I could add a test containing the examples from the README in another package to avoid this. What
do you think?

**Comment by sksamuel** [comment]
Yes that seems like a good idea.

**Comment by jeroentervoorde** [comment]
Alright. I'll do that tomorrow.

**Comment by jeroentervoorde** [comment]
I submitted a PR for the tests

**Comment by sksamuel** [comment]
All fixed now in 1.5.1

**Comment by dileepprasad-developer** [comment]
I was trying the same with 1.6.3 and got below error

Error:(102, 49) Could not find implicit SchemaFor[Composer]
    val output = AvroOutputStream.json[Composer](baos)
Error:(102, 49) not enough arguments for method json: (implicit evidence$5: com.sksamuel.avro4s.SchemaFor[Composer], implicit evidence$6: com.sksamuel.avro4s.ToRecord[Composer])com.sksamuel.avro4s.AvroJsonOutputStream[Composer].
Unspecified value parameters evidence$5, evidence$6.
    val output = AvroOutputStream.json[Composer](baos)
